### PR TITLE
Log exceptions thrown by explicit instrumentation

### LINKED ISSUE #74: Log exceptions from explicit instrumentations added with @Traced
https://github.com/eclipse/microprofile-opentracing/blob/master/spec/src/main/asciidoc/microprofile-opentracing.asciidoc#321-the-traced-annotation

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #74 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: tck/src/main/java/org/eclipse/microprofile/opentracing/tck/tracer/TestSpan.java [hunk 1]
```diff
@@ -392,12 +392,27 @@ public boolean equals(final Object obj) {
...
+                for (Map.Entry<String, ?> logEntry: logFieldsX.entrySet()) {
+                    Object valY = logFieldsY.get(logEntry.getKey());
+                    if (valY == null) {
+                        System.err.printf("Log field %s not present in the logs\n", logEntry.getKey());
+                        return false;
+                    }
+
+                    if (!logEntry.getKey().equals("error.object")) {
+                        if (!valY.equals(logEntry.getValue())) {
+                            System.err.printf("Log values no not match key=%s, %s, %s\n\n\n\n", logEntry.getKey(), logEntry.getValue(), valY);
```
Comment (objectiser) [inline tck/src/main/java/org/eclipse/microprofile/opentracing/tck/tracer/TestSpan.java]: Type 'no' should be 'do'

File: tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingBaseTests.java [hunk 2]
```diff
@@ -226,18 +226,39 @@ protected void assertEqualTrees(ConsumableTree<TestSpan> returnedTree,
...
+                        && !key.equals(Tags.ERROR.getKey()));
+            }
+        });
 
         // It's okay if the returnedTree has log entries other than the ones we
         // want to compare, so just remove those
-        returnedTree.visitTree(span -> span.getLogEntries()
-            .removeIf(logEntry -> true));
+        returnedTree.visitTree(span -> {
+            System.out.println(span.getLogEntries().size() + "  " + span.getLogEntries());
```
[the remark this answers is not in the document]
Comment (pavolloffay) [reply tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingBaseTests.java]: I will remove these log outputs

**Comment by pavolloffay** [comment]
@fmhwong could you please also have a look?

**Comment by pavolloffay** [comment]
@fmhwong I will merge to unblock other PRs. Please comment even if it's merged I will handle it in a separate PR if necessary.

**Comment by fmhwong** [comment]
LGTM

**Pull request merged**